File: Javascript/TwoSum.js
/**
 * Solution 1
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const result = {}

    for (let i = 0; i < nums.length; i++) {
        let temp = target - nums[i]

        if (result.hasOwnProperty(temp)) return [result[temp], i]
        result[nums[i]] = i
    }
};

/**
 * Solution 2
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let store = new Map();
    let result = [];

    nums.forEach((item, index) => {
        if (store.has(item)) {
            result.push(store.get(item), index);
        } else {
            let diff= target -item;
            store.set(diff, index);
        }
    })
    return result;
};

File: C#/TwoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> result = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++) {
            int j = target - nums[i];

            if (result.ContainsKey(j)) {
                return new []{result[j], i};
            }
            else {
                result[nums[i]] = i;
            }
            
        }

        return null;
    }
}